samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -82,7 +83,40 @@ def _execute_action(self, payload: str):
             elif boto_utils.get_client_error_code(client_ex) == "InvalidRequestContentException":
                 raise InvalidResourceBotoParameterException(client_ex) from client_ex
             raise ErrorBotoApiCallException(client_ex) from client_ex
-        return response
+
+    @abstractmethod
+    def _execute_lambda_invoke(self, payload: str):
+        pass
+
+
+class LambdaInvokeExecutor(AbstractLambdaInvokeExecutor):
+    """
+    Calls "invoke" method of "lambda" service with given input.
+    """
+
+    def _execute_lambda_invoke(self, payload: str) -> dict:
+        LOG.debug(
+            "Calling lambda_client.invoke with FunctionName:%s, Payload:%s, parameters:%s",
+            self._function_name,
+            payload,
+            self.request_parameters,
+        )
+        return cast(dict, self._lambda_client.invoke(**self.request_parameters))
+
+
+class LambdaInvokeWithResponseStreamExecutor(AbstractLambdaInvokeExecutor):
+    """
+    Calls "invoke_with_response_stream" method of "lambda" service with given input.
+    """
+
+    def _execute_lambda_invoke(self, payload: str) -> dict:
+        LOG.debug(
+            "Calling lambda_client.invoke_with_response_stream with FunctionName:%s, Payload:%s, parameters:%s",
+            self._function_name,
+            payload,
+            self.request_parameters,
+        )
+        return cast(dict, self._lambda_client.invoke_with_response_stream(**self.request_parameters))
```

Comment:
It does but it is not clear since typing is not sorted here. 
As @hawflau suggested, I will start using lambda client from mypy, let me see if it fixes this casting requirement.